dax/spiders.py

```diff
@@ -0,0 +1,518 @@
+"""
+    Title: spiders.py
+    Author: Benjamin Yvernault
+    contact: b.yvernault@ucl.ac.uk
+    Purpose:
+        Spider base class and class for Scan and Session spider
+        Spider name must be: Spider_[name]_v[version].py
+        Utils for spiders
+"""
+
+__author__ = 'Benjamin Yvernault'
+__email__ = 'b.yvernault@ucl.ac.uk'
+__purpose__ = "Spider base class, Scan and Session spider class, and Utils for spiders."
+__version__ = '1.0.0'
+__modifications__ = '26 August 2015 - Original write'
+
+import os
+import re
+import sys
+import time
+import getpass
+import collections
+from dax import XnatUtils
+from datetime import datetime
+
+class Spider(object):
+    """
+        Base class for spider
+
+        :param spider_path: spider file path
+        :param jobdir: directory for temporary files
+        :param xnat_project: project ID on XNAT
+        :param xnat_subject: subject label on XNAT
+        :param xnat_session: experiment label on XNAT
+        :param xnat_host: host for XNAT if not set in environment variables
+        :param xnat_user: user for XNAT if not set in environment variables
+        :param xnat_pass: password for XNAT if not set in environment variables
+        :param suffix: suffix to the assessor creation
+    """
+    def __init__(self, spider_path, jobdir, xnat_project, xnat_subject, xnat_session,
+                 xnat_host=None, xnat_user=None, xnat_pass=None, suffix=""):
+        # Spider path:
+        self.spider_path = spider_path
+        # directory for temporary files + create it
+        self.jobdir = XnatUtils.makedir(os.path.abspath(jobdir))
+        # to copy results at the end
+        self.spider_handler = None
+        # Xnat info:
+        self.xnat_project = xnat_project
+        self.xnat_subject = xnat_subject
+        self.xnat_session = xnat_session
+        # Xnat connection settings:
+        self.host = self.get_default_value("host", "XNAT_HOST", xnat_host)
+        self.user = self.get_default_value("user", "XNAT_USER", xnat_user)
+        self.pwd = self.get_pwd(xnat_pass, xnat_user)
+        # Suffix
+        if suffix[0] != '_': self.suffix = '_'+suffix
```

Comment:
You could make it
if suffix and suffix[0] != '_'

if '' evaluates to false.

On Wed, Sep 30, 2015 at 11:30 AM, Brian D. Boyd <notifications@github.com>
wrote:

> In dax/spiders.py
> <https://github.com/VUIIS/dax/pull/21#discussion_r40816898>:
>
> > +        # Spider path:
> > +        self.spider_path = spider_path
> > +        # directory for temporary files + create it
> > +        self.jobdir = XnatUtils.makedir(os.path.abspath(jobdir))
> > +        # to copy results at the end
> > +        self.spider_handler = None
> > +        # Xnat info:
> > +        self.xnat_project = xnat_project
> > +        self.xnat_subject = xnat_subject
> > +        self.xnat_session = xnat_session
> > +        # Xnat connection settings:
> > +        self.host = self.get_default_value("host", "XNAT_HOST", xnat_host)
> > +        self.user = self.get_default_value("user", "XNAT_USER", xnat_user)
> > +        self.pwd = self.get_pwd(xnat_pass, xnat_user)
> > +        # Suffix
> > +        if suffix[0] != '_': self.suffix = '_'+suffix
>
> what if suffix is empty string?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/VUIIS/dax/pull/21/files#r40816898>.
>